compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -56,46 +56,374 @@ namespace TR { class Register; }
+   }
+
+static void fillFieldBFA(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0x7) == val, "0x%x is out-of-range for BFA field", val);
+   *cursor |= val << 18;
+   }
+
+static void fillFieldBFB(TR::Instruction *instr, uint32_t *cursor, TR::RealRegister *reg)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, reg, "Attempt to fill BFB field with null register");
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, reg->getKind() == TR_CCR, "Attempt to fill BFB field with %s, which is not a CCR", reg->getRegisterName(instr->cg()->comp()));
+   reg->setRegisterFieldRB(cursor);
+   }
+
+static void fillFieldU(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0xfu) == val, "0x%x is out-of-range for U field", val);
+   *cursor |= val << 12;
+   }
+
+static void fillFieldBFW(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0xfu) == val, "0x%x is out-of-range for BF/W field", val);
+   *cursor |= ((val ^ 0x8) & 0x8) << 13;
+   *cursor |= (val & 0x7) << 23;
+   }
+
+static void fillFieldFLM(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0xffu) == val, "0x%x is out-of-range for FLM field", val);
+   *cursor |= val << 17;
+   }
+
+static void fillFieldFXM(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0xffu) == val, "0x%x is out-of-range for FXM field", val);
+   *cursor |= val << 12;
+   }
+
+static void fillFieldFXM1(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, populationCount(val), "0x%x is invalid for FXM field, expecting exactly 1 bit set", val);
+   fillFieldFXM(instr, cursor, val);
+   }
+
+static void fillFieldDCM(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0x3fu) == val, "0x%x is out-of-range for DCM/DQM field", val);
+   *cursor |= val << 10;
+   }
+
+static void fillFieldSH5(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0x1fu) == val, "0x%x is out-of-range for SH(5) field", val);
+   *cursor |= val << 11;
+   }
+
+static void fillFieldSH6(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0x3fu) == val, "0x%x is out-of-range for SH(6) field", val);
+   *cursor |= (val & 0x1fu) << 11;
+   *cursor |= (val & 0x20u) >> 4;
+   }
+
+static void fillFieldSIM(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, isValidInSignExtendedField(val, 0x1fu), "0x%x is out-of-range for SIM field", val);
+   *cursor |= (val & 0x1f) << 16;
+   }
+
+static void fillFieldSI(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
```

Comment:
Ah, you're right that this name should include the size since we already have `SI5` fields and this is needlessly ambiguous. Fixed in a3245c9578b739d36ee5ab8b23bf0a398fb7a2c7.